src/main/java/pixelperfect/Spaceship.java

```diff
@@ -15,6 +16,7 @@
 public class Spaceship {
 
 	private static final Spaceship ship = new Spaceship();
+	private RouteGenerator routeGenerator;
```

Comment:
I don't agree totaly: I think Route should know that it is a Route. The RouteGenerator determines how the Route should look like. (keeping jobs apart)